app/models/bank_transaction.rb

```diff
@@ -60,7 +60,7 @@ def bind_invoice(invoice_no, manual: false)
 
     invoice = Invoice.find_by(number: invoice_no)
     errors.add(:base, I18n.t('invoice_was_not_found')) unless invoice
```

Comment:
Let's consider to move this check also to` #validate_invoice_data(invoice)` method.